aiogram/contrib/fsm_storage/redis.py

```diff
@@ -244,49 +386,59 @@ def __init__(self, host: str = 'localhost', port=6379, db=None, password=None,
         self._data_ttl = data_ttl
         self._bucket_ttl = bucket_ttl
 
-        self._redis: typing.Optional[aioredis.RedisConnection] = None
+        self._redis: typing.Optional[AioRedisAdapterBase] = None
         self._connection_lock = asyncio.Lock(loop=self._loop)
 
+    @deprecated("This method will be removed in aiogram v3.0. "
+                "You should use your own instance of Redis.", stacklevel=3)
     async def redis(self) -> aioredis.Redis:
-        """
-        Get Redis connection
-        """
-        # Use thread-safe asyncio Lock because this method without that is not safe
-        async with self._connection_lock:
-            if self._redis is None or self._redis.closed:
-                self._redis = await aioredis.create_redis_pool((self._host, self._port),
-                                                               db=self._db, password=self._password, ssl=self._ssl,
-                                                               minsize=1, maxsize=self._pool_size,
-                                                               loop=self._loop, **self._kwargs)
+        adapter = await self._get_adapter()
+        return await adapter.get_redis()
+
+    async def _get_adapter(self) -> AioRedisAdapterBase:
+        """Get adapter based on aioredis version."""
+        if self._redis is None:
+            redis_version = aioredis.__version__.split(".")[0]
+            connection_data = dict(
+                host=self._host,
+                port=self._port,
+                db=self._db,
+                password=self._password,
+                ssl=self._ssl,
+                pool_size=self._pool_size,
+                loop=self._loop,
+                **self._kwargs,
+            )
+            if redis_version == 1:
+                self._redis = AioRedisAdapterV1(**connection_data)
+            elif redis_version == 2:
```

Comment:
You're comparing `str` with `int`
```suggestion
            if redis_version == "1":
                self._redis = AioRedisAdapterV1(**connection_data)
            elif redis_version == "2":
```